tests/components/synology_dsm/test_backup.py

```diff
@@ -0,0 +1,542 @@
+        homeassistant_version="2024.12.0",
+        name="Test",
+        protected=True,
+        size=0,
+    )
+
+    with (
+        patch(
+            "homeassistant.components.backup.manager.BackupManager.async_get_backup",
+        ) as fetch_backup,
+        patch(
+            "homeassistant.components.backup.manager.read_backup",
+            return_value=test_backup,
+        ),
+        patch("pathlib.Path.open") as mocked_open,
+    ):
+        mocked_open.return_value.read = Mock(side_effect=[b"test", b""])
+        fetch_backup.return_value = test_backup
+        resp = await client.post(
+            "/api/backup/upload?agent_id=synology_dsm.Mock Title",
+            data={"file": StringIO("test")},
+        )
+
+    assert resp.status == 201
+    assert f"Uploading backup {backup_id}" in caplog.text
+    mock: AsyncMock = setup_dsm_with_filestation.file.upload_file
+    assert len(mock.mock_calls) == 2
+    assert mock.call_args_list[0].kwargs["filename"] == "test-backup.tar"
+    assert mock.call_args_list[0].kwargs["path"] == "/ha_backup/my_backup_path"
+    assert mock.call_args_list[1].kwargs["filename"] == "test-backup_meta.json"
+    assert mock.call_args_list[1].kwargs["path"] == "/ha_backup/my_backup_path"
+
+
+async def test_agents_upload_error(
+    hass: HomeAssistant,
+    hass_client: ClientSessionGenerator,
+    caplog: pytest.LogCaptureFixture,
+    setup_dsm_with_filestation: MagicMock,
+) -> None:
+    """Test agent error while uploading backup."""
+    client = await hass_client()
+    backup_id = "test-backup"
+    test_backup = AgentBackup(
+        addons=[AddonInfo(name="Test", slug="test", version="1.0.0")],
+        backup_id=backup_id,
+        database_included=True,
+        date="1970-01-01T00:00:00.000Z",
+        extra_metadata={},
+        folders=[Folder.MEDIA, Folder.SHARE],
+        homeassistant_included=True,
+        homeassistant_version="2024.12.0",
+        name="Test",
+        protected=True,
+        size=0,
+    )
+
+    with (
+        patch(
+            "homeassistant.components.backup.manager.BackupManager.async_get_backup",
+        ) as fetch_backup,
+        patch(
+            "homeassistant.components.backup.manager.read_backup",
+            return_value=test_backup,
+        ),
+        patch("pathlib.Path.open") as mocked_open,
+    ):
+        mocked_open.return_value.read = Mock(side_effect=[b"test", b""])
+        fetch_backup.return_value = test_backup
+        setup_dsm_with_filestation.file.upload_file.side_effect = (
+            SynologyDSMAPIErrorException("api", "500", "error")
+        )
+        resp = await client.post(
+            "/api/backup/upload?agent_id=synology_dsm.Mock Title",
+            data={"file": StringIO("test")},
+        )
+
+    assert resp.status == 201
+    assert f"Uploading backup {backup_id}" in caplog.text
+    assert (
+        "Error uploading to synology_dsm.Mock Title: Failed to upload backup"
+        in caplog.text
+    )
```

Comment:
not sure, why it still looks like the upload was OK, even we clearly get a `BackupAgentError` which results in an error message in the log